### PR TITLE
add PostHog

### PR DESCRIPTION
**Pull request by AndrMoura**
First of all, thank you for your contribution! 😄

Please note that this repo is mostly for popular apps with thousands of stars and tens of thousands of downloads. If you'd like to add a less popular app, you can always [create your own 3rd party repo](https://github.com/caprover/one-click-apps#build-your-own-one-click-app-repository) and add your app there.

### ☑️ Self Check before Merge

- [x] I have tested the template using the method described in README.md thoroughly
- [x] I have ensured that I put as much default values as possible (except passwords) to ensure minimum effort required for end users to get started.
- [x] I have ensured that I am not using the "latest" tag as this tag is dynamically changing and might break the one-click app. Use a fixed version.
- [x] I have made sure that instructions.start and instructions.end are clear and self-explanatory.
- [x] Icon is added as a png file to the logos directory.
- [x] I've executed the checks if necessary by running `npm ci && npm run validate_apps && npm run formatter` (If failling run the prettier: `npm run formatter-write`)
- [x] I will take responsibility addressing any issues that arises as a result of this PR (maintaining this app).

### CODE REVIEW

File: public/v4/apps/posthog.yml [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+            - $$cap_appname-zookeeper-datalog:/datalog
+            - $$cap_appname-zookeeper-data:/data
+            - $$cap_appname-zookeeper-logs:/logs
+
+    $$cap_appname-clickhouse:
+        restart: on-failure
+        caproverExtra:
+            notExposeAsWebApp: 'true'
+            dockerfileLines:
+                - FROM andremoura/helper AS helper
```
Comment (githubsaturn) [inline public/v4/apps/posthog.yml:43]: This has an implicit `latest` tag, can you please ensure a concrete version here?

File: public/v4/apps/posthog.yml [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+        - id: $$cap_enable_elastic_search
+          description: Enable elastic search with Temporal
+          label: Enable Elastic Search
+          defaultValue: 'false'
+          validRegex: /^(true|false)$/
+
+    instructions:
+        start: >-
+            Enter your PostHog Configuration parameters and click on next. The process will take a just a few minutes to finish.
```
Comment (githubsaturn) [inline public/v4/apps/posthog.yml:323]: This is one of the most complex apps that we have in the repo. Can you please add a suggested minimum hardware requirement for this suite of apps?

Also, maybe add a note saying that "this one click app deploys **16 distinct apps** and it requires ....."

**Review by githubsaturn — COMMENTED** [review]
This is probably the most complex one click app that we have on CapRover! What a beast!

File: public/v4/apps/posthog.yml [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+            - $$cap_appname-zookeeper-datalog:/datalog
+            - $$cap_appname-zookeeper-data:/data
+            - $$cap_appname-zookeeper-logs:/logs
+
+    $$cap_appname-clickhouse:
+        restart: on-failure
+        caproverExtra:
+            notExposeAsWebApp: 'true'
+            dockerfileLines:
+                - FROM andremoura/helper:1.0 AS helper
```
Comment (githubsaturn) [inline public/v4/apps/posthog.yml:43]: I just realized that temporary image for some files, why not inline everything here?
https://github.com/AndrMoura/posthog-caprover/blob/main/helper_files.sh

Something like this:
```
FROM alpine AS build 
WORKDIR /
RUN curl -L https://github.com/PostHog/posthog/tarball/c25b571388a9ad05d529a14daee8f49819280496 | tar -xz --wildcards --strip-components=2 -C "./" '*/posthog/idl/*'
RUN curl -L https://github.com/PostHog/posthog/tarball/c25b571388a9ad05d529a14daee8f49819280496 | tar -xz --wildcards --strip-components=2 -C "$DOCKER_DIR" '*/docker/*'

Comment (AndrMoura) [reply public/v4/apps/posthog.yml]: I over-complicated things a bit there! Fixed! :)

**Comment by githubsaturn** [comment]
Nice!

**Pull request merged**